packages/bitcore-wallet-client/src/lib/common/utils.ts

```diff
@@ -522,6 +522,12 @@ export class Utils {
     const suffix = Constants.EVM_CHAINSUFFIXMAP[chain.toLowerCase()];
     const coinIsAChain = !!Constants.EVM_CHAINSUFFIXMAP[coin.toLowerCase()];
     if (suffix && (coinIsAChain || chain.toLowerCase() !== 'eth')) {
+       // Special handling for usdc.e and usdc on matic
+      if (coin.toLowerCase() === 'usdc.e') {
+        return 'USDC_m';
+      } else if (coin.toLowerCase() === 'usdc') {
+        return 'USDCn_m';
```

Comment:
hard code suffix?

Reply:
I think it's fine because it's for matic and those particular coins and we don't have other chains apart from eth and matic. But I could add a chain check.